### DISCUSSION TITLE
Mapping fields on Forms/Integration_Base not updated on already added forms

### DISCUSSION THREAD

**Discussion by stefanolissa** [elementor/elementor#25644] [Developers Community]
### Start a discussion

Hi,
using the Forms/Integration_Base to add an "Action after submit" setting, if the mapping fields configuration (returned by `get_fields_map_control_options()`) changes (for example because a new field is added), forms already added to a page don't show the new field, while if I add a new form, the new field is shown. It seems a form caches the initial mapping settings forever. Even if the action is removed, the page saved, and added once again, the old field list is shown.

It happens only to field mapping controls (added `using register_fields_map_control()`), while other controls in the section, when changed, are shown.

On the method `get_fields_map_control_options()` I'm returning an array with the list of fields under the "default" key:

```
protected function get_fields_map_control_options() {
        $options = [
            'default' => [
                [
                    'remote_id' => 'consent',
                    'remote_label' => 'Consent checkbox',
                    'remote_type' => 'text'
                ],
                ....
        ]];

return $options;
}
```

Isn't it the correct way to return the field options? 

Stefano.

**Comment by ludekcerny** [comment]
Hi, Stefano, 

You are not the only one who is experiencing this behavior, I also found this problem https://github.com/elementor/elementor/issues/8938 with a suggested solution. 
In my case I solved it by manually changing _elementor_data directly in the database. 
For a one-time solution it works, but I would like to know how to solve it properly.
If you figure something out, please let me know. 

Ludek
